rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromReturnNewRector/Fixture/return_anonymous_class_implements.php.inc

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromReturnNewRector\Fixture;
+
+use Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromReturnNewRector\Source\SomeResponseInterface;
+
+final class ReturnAnonymousClassImplements
+{
+    public function action()
+    {
+        return new class implements SomeResponseInterface {};
+    }
+}
+
+?>
+-----
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromReturnNewRector\Fixture;
+
+use Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromReturnNewRector\Source\SomeResponseInterface;
+
+final class ReturnAnonymousClassImplements
+{
+    public function action(): \Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromReturnNewRector\Source\SomeResponseInterface
+    {
+        return new class implements SomeResponseInterface {};
```

Comment:
Nice, what about a case which mixes extends and implements?

Reply:
this only handle 1 type of parent, covered at 

https://github.com/rectorphp/rector-src/blob/625fcccc475c8a6b3bde49e5e64e66354ed4299a/packages/PHPStanStaticTypeMapper/TypeMapper/ObjectWithoutClassTypeMapper.php#L52-L62

for multi parents, that will fallback to `object`

Reply:
ok cool. we should cover it with a test

Reply:
I added implements multi interface fixture https://github.com/rectorphp/rector-src/pull/4669/commits/86e0a0537f27251ef855691d723fc36a2a96c05a

Reply:
thanks!